src/main/java/io/github/syst3ms/skriptparser/util/TimeUtils.java

```diff
@@ -0,0 +1,158 @@
+package io.github.syst3ms.skriptparser.util;
+
+import org.jetbrains.annotations.Nullable;
+
+import java.time.Duration;
+import java.util.Arrays;
+
+public class TimeUtils {
+
+    @Nullable
+    public static Duration parseDuration(String str) {
+        // TODO parsing gets in trouble with lists
+        if (str.isEmpty())
+            return null;
+        Duration dur = Duration.ZERO;
+
+        // If there is a ', and' in the string, like '5 days, and 3 seconds',
+        // then we return null since we need to keep consistency with lists.
+        if (str.contains(", and")) return null;
+
+        // Removing all ',' except the last one.
+        boolean endsComma = str.endsWith(",");
+        str = str.replaceAll(",", "");
+        if (endsComma) str = str.concat(",");
+
+        String[] split = str.split("\\s+");
+
+        // Days, hours, minutes, seconds, milliseconds. This is to keep track that they come in the right order.
+        // Like '5 days and 3 minutes', not '3 seconds and 5 hours'.
+        boolean[] passed = {false, false, false, false, false};
+
+        // Checks how many 'and' occurrences we had. We can only use 'and' once, before our last part.
+        // This is, again, to keep consistency with lists.
+        int currentAnd = 0;
+
+        for (int i = 0; i < split.length; i++) {
+            String s = split[i];
+
+            if (s.equals("and")) {
+                currentAnd++;
+                if ((split.length > 2 && !split[split.length - 3].equals("and")) || currentAnd > 1)
+                    return null;
+                continue;
+            }
+
+            long amount;
+
+            try {
+                amount = Long.parseLong(s);
+                s = split[++i];
+            } catch (NumberFormatException | ArrayIndexOutOfBoundsException ex) {
+                return null;
+            }
+
+            if (s.matches("days?")) {
+                if (passed[0] || oneIsTrue(passed[1], passed[2], passed[3], passed[4]))
+                    return null; // Days was already used elsewhere and is used again.
+                dur = dur.plusDays(amount);
+                passed[0] = true;
+            }
+
+            else if (s.matches("hours?")) {
```

Comment:
`else if`/`else`s should be on the same line as the closing bracket of the previous `if`/`else if`.